NDTensors/test/device_list.jl

```diff
@@ -0,0 +1,21 @@
+function devices_list(test_args)
+  devs = Vector{Function}(undef, 0)
+  if isempty(test_args) || "base" in test_args
+    push!(devs, NDTensors.cpu)
+  end
+
+  if "cuda" in test_args || "all" in test_args
+    CUDA.allowscalar()
+    if CUDA.functional()
```

Comment:
Maybe we should output a warning if it isn't functional?

Reply:
I think I would be surprised if I tried to run the CUDA backend tests but they just silently didn't run since CUDA wasn't set up properly.